gslib/cloud_api_delegator.py

```diff
@@ -209,6 +210,13 @@ def GetApiSelector(self, provider=None):
     elif self.prefer_api in (
         self.api_map[ApiMapConstants.SUPPORT_MAP][selected_provider]):
       api = self.prefer_api
+
+    if (api == ApiSelector.XML and context_config.get_context_config() and
+        context_config.get_context_config().use_client_certificate):
+      raise ArgumentException(
+          'User enabled mTLS by setting "use_client_certificate", but mTLS'
```

Comment:
Nit: You can add an actionable step here like you do for the error raised from GcsJsonApi

Reply:
Done